dashboard/app/models/user.rb

```diff
@@ -2424,6 +2430,13 @@ def self.from_omniauth(auth, params, request = nil)
     unless omniauth_user
       omniauth_user = create
       initialize_new_oauth_user(omniauth_user, auth, params)
+      klass_map = {
+        ::User::TYPE_TEACHER => ::Teacher,
+        ::User::TYPE_STUDENT => ::Student,
+        'staff' => ::Teacher # Powerschool sends through 'staff' instead of 'teacher'
+      }
+      klass = klass_map.fetch(omniauth_user.user_type, ::User)
+      omniauth_user = omniauth_user.becomes!(klass)
```

Comment:
Could we also move `klass_map` to `User.find_sti_class` and use the method instead?
```ruby
class User
    STI_CLASS_MAP = {
      TYPE_TEACHER => ::Teacher,
      TYPE_STUDENT => ::Student,
      'staff' => ::Teacher # Powerschool sends through 'staff' instead of 'teacher'
    }

    def self.find_sti_class(type_name)
      STI_CLASS_MAP[type_name]
    end
end
```

```suggestion
      sti_class = find_sti_class(omniauth_user.user_type)
      omniauth_user = omniauth_user.becomes!(sti_class) if sti_class
```

Reply:
Thanks for the suggestion, much cleaner! I've applied the updates.